exetera/core/dataframe.py

```diff
@@ -588,9 +590,9 @@ def _write_groupby_keys(self, ddf: DataFrame, write_keys=True):
                 
                 if self._sorted_index is not None:                    
                     field.apply_index(self._sorted_index, target=newfld)                  
-                    newfld.apply_filter(self._spans[:-1], in_place=True)
+                    newfld.apply_index(self._spans[:-1], in_place=True)
```

Comment:
Isn't this unrelated functionality? Looks like a bug in the groupby, which I'm guessing wasn't caught by unit tests. It really should have some unit testing associated with it

Reply:
yeah, I think it's the newly added unit tests in this PR that detected the bug.

Reply:
Before since apply_filter didn't check if the parameter of "filter_to_apply" is boolean or not. It means when the input "filter_to_apply"  is an index (e.g. [0,1,3,6] here), apply_filter still works fine in the code below, which means **apply_filter worked the same as  apply_index before**.
``
dest_data = source.data[:][filter_to_apply_]
``

After we update the input to boolean type for apply_filter, then "filter_to_apply" is update from [0,1,3,6] to [False, True, True, True]. And the code above works as filter, instead of index. And that's why apply_filter didn't work anymore, so have to be updated to apply_index here.